include/envoy/local_info/local_info.h

```diff
@@ -25,17 +25,17 @@ class LocalInfo {
   /**
    * Human readable zone name. E.g., "us-east-1a".
    */
-  virtual const std::string& zoneName() const PURE;
+  virtual const std::string zoneName() const PURE;
```

Comment:
This does have some perf implication (not huge but not zero either). Is this really necessary? I'm not sure we can rely on compiler to optimize this.